### ISSUE TITLE
LightGBMRegressor can raise ValueError: DataFrame.dtypes for data must be int, float or bool. when BooleanNullable type passed in via X

### ISSUE THREAD

**Issue by tamargrey** [alteryx/evalml#3925]
This was first discovered in [this commit](https://github.com/alteryx/evalml/pull/3895/commits/cb4e1ee866d40dd5cebe274904cfd50880d1193b), which forced nullable type usage into AutoMLSearch. It can be reproduced if nullable types are passed directly into the component:

```python
    X = pd.DataFrame(
        {"feature_1": [True, False, False, True, True, False], "feature_2": ["a", "a", "b", "b", "c", "c"]},
    )
    X.ww.init(logical_types={"feature_1": "BooleanNullable", "feature_2": "categorical"})
    y = pd.Series([1, 1, 0, 0, 0, 1])

    clf = LightGBMRegressor()
    clf.fit(X, y)
```

Note - we have handling for IntegerNullable types in this component, so only the BooleanNullable case fails. Additionally, this will not currently be seen in automl search because of the replace nullable types component, but we should consider adding nullable handling into this component class itself so that it can independently support nullable types.

**Comment by tamargrey** [comment]
Should be fixed by the future lightgbm 4.0.0 release: https://github.com/microsoft/LightGBM/issues/5153

**Comment by tamargrey** [comment]
duplicate of https://github.com/alteryx/evalml/issues/3924 - closing
